### ISSUE TITLE
Empty body causes "null" content from wrap-json-response

### ISSUE THREAD

**Issue by joelittlejohn** [ngrunwald/ring-middleware-format#39]
When a response that has an empty body passes through wrap-json-response, the middleware parses a missing body to the json string `"null"` and produces a content length of 4 bytes.

```
app.web=> (use 'ring.middleware.format-response)
nil

app.web=> ((wrap-json-response identity) {:status 204})
{:headers {"Content-Length" "4", "Content-Type" "application/json; charset=utf-8"}, :body #<BufferedInputStream java.io.BufferedInputStream@7175bbbd>, :status 204}

app.web=> (-> ((wrap-json-response identity) {:status 204}) (update-in [:body] slurp))
{:headers {"Content-Length" "4", "Content-Type" "application/json; charset=utf-8"}, :body "null", :status 204}
```

I would expect the middleware to leave this response alone.

**Comment by joelittlejohn** [comment]
I'm using ring-json as an alternative now, but I thought I'd raise this to get some comments from elsewhere too.

**Comment by hlship** [comment]
I'm surprised that this was only reported 6 hours ago ... I've been meaning to report it for months.

I really think this just needs a little wrapper to ignore a nil body.

**Comment by hlship** [comment]
Created a PR: https://github.com/ngrunwald/ring-middleware-format/pull/40

**Comment by ngrunwald** [comment]
Closed by #40
